### PR TITLE
CBL-951 Add database delete/close tests in URL Endpoint Listener, TestClientCertAuthRootCertsError, and code clean up.

### PR DESCRIPTION
**Pull request by Sandychuang8**
CBL-951 Add close/delete database tests in url endpoint listener. Also add unit test case that is added in iOS.

### CODE REVIEW

File: src/Couchbase.Lite.Tests.Shared/URLEndpointListenerTest.cs [hunk 1]
```diff
@@ -402,6 +440,50 @@ public void TestClientCertAuthenticatorRootCerts()
             _listener.Stop();
         }
 
+        //[Fact]
```
Comment (borrrden) [inline src/Couchbase.Lite.Tests.Shared/URLEndpointListenerTest.cs:443]: Is there a reason this test is added just to be disabled?

**Review by borrrden — CHANGES_REQUESTED** [review]
Please edit the title ~and description~ of the PR to something meaningful

**Comment by Sandychuang8** [comment]
All tests now passed besides Linux is throwing (int) CouchbaseLiteError.WebSocketGoingAway instead of (int)CouchbaseLiteError.WebSocketUserPermanent in TestP2PPermanentFailureDuringSend 
I created a separate issue to track (https://issues.couchbase.com/browse/CBL-1255)

File: src/Couchbase.Lite.Tests.Shared/URLEndpointListenerTest.cs [hunk 2]
```diff
@@ -634,7 +672,7 @@ public void TestReplicatorAndListenerOnSameDatabase()
 
             repl1.Start();
             repl2.Start();
-            WaitHandle.WaitAll(new[] {wait1.WaitHandle, wait2.WaitHandle}, _timeout)
+            WaitHandle.WaitAll(new[] {wait1.WaitHandle, wait2.WaitHandle}, TimeSpan.FromSeconds(20))
```
Comment (borrrden) [inline src/Couchbase.Lite.Tests.Shared/URLEndpointListenerTest.cs:675]: The point of assigning `_timeout` is to make things dynamic, just FYI

**Review by borrrden — APPROVED** [review]

File: src/Couchbase.Lite.Tests.Shared/URLEndpointListenerTest.cs (hunk 2 above)
Comment (Sandychuang8) [reply src/Couchbase.Lite.Tests.Shared/URLEndpointListenerTest.cs:675]: yeah, the default value is 15 sec... Just want to be safe to wait longer for two separate replicators to stop.

**Pull request merged**